shopinvader_guest_mode/services/guest_service.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.component.core import Component
+from werkzeug.exceptions import NotFound, Forbidden
+
+
+class GuestService(Component):
+    _name = 'shopinvader.guest.service'
+    _inherit = 'shopinvader.customer.service'
+    _usage = 'guest'
+
+    # The following method are 'public' and can be called from the controller.
+    def create(self, **params):
+        if not self.shopinvader_backend.is_guest_mode_allowed:
+            raise Forbidden('Guest mode not allowed.')
+        params['is_guest'] = True
+        self._archive_existing_binding(params['email'])
+        resp = super(GuestService, self).create(**params)
+        resp['store_cache']['customer']['is_guest'] = True
```

Comment:
Maybe we can move the "is_guest" information directly in the mapping? So we will always have this key with False or True in the customer variable.